src/components/Register.js

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+const BASE_URL = '/'
+
+
+
+const Register = (props) => {
+    const { setUser, setToken } = props;
+    const [firstName, setFirstName] = useState('');
+    const [lastName, setLastName] = useState('');
+    const [email, setEmail] = useState('');
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+    const [imageURL, setImageURL] = useState('');
+
+    
+    const signUp = async ({firstName, lastName, email, username, password, imageURL, isAdmin}) => {
+        try {
+            const response = await axios.post(`${ BASE_URL }api/users/register`, {firstName, lastName, email, username, password, imageURL, isAdmin})
```

Comment:
Might want to create the api call in its own function that lives on the app, and then gets passed down to the child components. Then you can set a status automatically if there is an error or success.